mongo.go

```diff
@@ -123,7 +123,12 @@ func (m Handler) Update(ctx context.Context, item *resource.Item, original *reso
 		return err
 	}
 	defer m.close(c)
-	err = c.Update(bson.M{"_id": original.ID, "_etag": original.ETag}, mItem)
+
+	s := bson.M{"_id": original.ID}
+	if original.ETag != "" {
+		s["_etag"] = original.ETag
+	}
```

Comment:
Actually, what about adding:
```go
} else {
    s["_etag"] = bson.M{"$exists": false)
}
```
Would that work? 

If it does, I suspect that should satisfy @rs PATCH consistency concerns on the ticket. As if, when orginal.Etag is not set, if two people have fetched an instance and both are PATCHing or PUTing the old resource, one of them should fail.

Could probably be validated with a (manual) test?

Reply:
That is a very good idea. Thanks. Will add test for it in `mongo_test.go`.

Reply:
Looking at the `rest/method_item_update/delete.go` `original` is fetched from DB just before the corresponding Update/Delete operation. This check will not do much, except maybe fixing concurrent Update/Delete on the same document.

Reply:
4 should fail, if possible.

Reply:
Why it should fail?  One is not supplying `If-Match-None`, so hes is okay with possible overwrites.
And again, this is just for DB bootstrapping. Production system will place `_etag` on each item it creates/updates.

Reply:
Right.

Reply:
So, yes, 3 & 4 should both work when `If-Match` is not given. If `If-Match` _is_ given (either for both requests, or just for 4), 4 should fail.

Reply:
Yes, I did meant `If-Match` sorry.
If `If-Match` is given in 3 or 4, then the PUT will be successful only if ETags match.
I don't think there is any chance supplying `If-Match` to a item that has no `_etag` field in it. There wouldn't be a way to obtain that `If-Match` value in the first place.

Reply:
Would `IF-Match=""` work though?

Reply:
Depends a bit on the rest-layer code for checking the header, as `""` is the default string value in go...

Reply:
`If-Match` == "" is equivalent to not supplying it at all.

Reply:
Alright. So that means it won't work as a measure against data races😔 Unless we where to generate an E-tag during the firat GET operation, but that could potentially get heary.

Is there a way to make this configurable like @rs suggests in the ticket?

Reply:
@rs, any views?

Reply:
When the etag is missing we could produce a weak etag with the document id, something like "W/<doc id>". On update, when we get this kind of id, we use the `{$not: {$exists: "_etag"}}` condition.

Reply:
@rs your idea might work. Let me think a bit on it.